winml/lib/Api.Image/EventTimer.h

```diff
@@ -0,0 +1,29 @@
+#include "pch.h"
+
+class EventTimer
+{
+public:
+  EventTimer()
+  {
+  }
+  
+  bool Start()
+  {
+    auto now = std::chrono::high_resolution_clock::now();
+    if (!_started || 
+         std::chrono::duration_cast<std::chrono::microseconds>(now - _startTime).count() > _kDurationBetweenSendingEvents)
+    {
+      _started = true;
+      _startTime = std::chrono::high_resolution_clock::now();
+      return true;
+    }
+
+    return false;
+  }
+
+private:
+  bool _started = false;
+  std::chrono::steady_clock::time_point _startTime;
+  constexpr static long long _kDurationBetweenSendingEvents = 1000 * 50;  // duration in (us). send an Event every 50 ms;
```

Comment:
Or simply 50000?

Reply:
it's constexpr, so it resolves to the same thing. I find this easier to read how many microseconds it's doing in case we ever want to change it.